auth/test/test_credentials.py

```diff
@@ -156,7 +156,7 @@ def test_invalid_unicode_robot(app):
     result, kind = validate_credentials("devtable+somerobot", token)
     assert kind == CredentialKind.robot
     assert not result.auth_valid
-    msg = "Could not find robot with specified username"
+    msg = "Could not find robot with username: devtable+somerobot and supplied password."
```

Comment:
Why this change?

Reply:
https://github.com/quay/quay/blob/master/data/model/user.py#L429
We were previously ignoring non-ascii characters, in the token, which is why the `UnicodeEncodeError` block was skipped (the non-ascii quotes were being stripped from the byte string)

https://github.com/quay/quay/blob/python3/data/model/user.py#L431
Since we're dealing with unicode now, we just try to encode the token as ascii to ensure that it's just ascii characters, and (as expected) get the exception.

Reply:
Cool